api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -89,4 +125,13 @@ interface Result {
      */
     Iterable<String> orphanFileLocations();
   }
+
+  /**
+   * Defines the Delete Orphan files behaviour when there is mismatch in prefix(scheme/authority)
+   * ERROR - Throws an exception when prefix mismatch
+   * IGNORE - No action when prefix mismatch
+   */
+  enum PrefixMismatchMode {
+    ERROR, IGNORE
```

Comment:
I thought a bit more about this and maybe we do need `DELETE` mode, even though I was reluctant to support it initially. The behavior will be you get an exception with all conflicting schemes/authorities by default and are asked to resolve conflicts. If that's not enough, then you can use either IGNORE or DELETE to handle the remaining conflicts.